accord-core/src/test/java/accord/burn/random/FrequentLargeRange.java

```diff
@@ -18,59 +18,121 @@
 
 package accord.burn.random;
 
-import accord.utils.Invariants;
+import accord.utils.Gen;
+import accord.utils.Gen.LongGen;
+import accord.utils.Gens;
 import accord.utils.RandomSource;
 
-public class FrequentLargeRange implements RandomLong
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+public class FrequentLargeRange implements LongGen
 {
-    private final RandomLong small, large;
-    private final double ratio;
-    private final int steps;
-    private final double lower, upper;
-    private int run = -1;
-    private long smallCount = 0, largeCount = 0;
-
-    public FrequentLargeRange(RandomLong small, RandomLong large, double ratio)
+    private final LongGen small, large;
+    private final Gen<Boolean> runs;
+
+    public FrequentLargeRange(LongGen small, LongGen large, double ratio)
     {
-        Invariants.checkArgument(ratio > 0 && ratio <= 1);
         this.small = small;
         this.large = large;
-        this.ratio = ratio;
-        this.steps = (int) (1 / ratio);
-        this.lower = ratio * .8;
-        this.upper = ratio * 1.2;
+        this.runs = Gens.bools().runs(ratio);
```

Comment:
thought about this after re-reviewing the code; `steps` is really `maxRuns` and because its `1 / ratio` it means that less frequent events have much higher maxRuns... it wouldn't be hard to expose this as a config so we can have more consistent maxRuns.... `1 / .01 == 100`, which feels like too many runs....

The behavior is not changed in this patch, but just something I thought about while reviewing.

Reply:
Ideally this is something we would randomise for each run, but let's look at that in follow-up.